srcCpp/perftest_publisher.cxx

```diff
@@ -424,34 +424,25 @@ bool perftest_cpp::validate_input()
             _PM.set<unsigned long long>("executionTime", 60);
         }
         // Check if scan is large data or small data
-        if (scanList[0] < (unsigned long long)(std::min)
-                    (MAX_SYNCHRONOUS_SIZE, MAX_BOUNDED_SEQ_SIZE)
-                && scanList[scanList.size() - 1] > (unsigned long long)(std::min)
-                    (MAX_SYNCHRONOUS_SIZE, MAX_BOUNDED_SEQ_SIZE)) {
+        if (scanList[0] < (unsigned long long) MAX_BOUNDED_SEQ_SIZE
+                && scanList[scanList.size() - 1] 
+                    > (unsigned long long) MAX_BOUNDED_SEQ_SIZE) {
             fprintf(stderr, "The sizes of -scan [");
             for (unsigned int i = 0; i < scanList.size(); i++) {
                 fprintf(stderr, "%llu ", scanList[i]);
             }
             fprintf(stderr,
                     "] should be either all smaller or all bigger than %d.\n",
-                    (std::min)(MAX_SYNCHRONOUS_SIZE, MAX_BOUNDED_SEQ_SIZE));
+                    MAX_BOUNDED_SEQ_SIZE);
             return false;
         }
     }
 
-    // Check if we need to enable Large Data. This works also for -scan
-    if (_PM.get<unsigned long long>("dataLen") > (unsigned long long) (std::min)(
-            MAX_SYNCHRONOUS_SIZE,
-            MAX_BOUNDED_SEQ_SIZE)) {
+    // Check if we need to enable the use of unbounded sequences. This works also for -scan
+    if (_PM.get<unsigned long long>("dataLen") > MAX_BOUNDED_SEQ_SIZE) {
         if (_PM.get<int>("unbounded") == 0) {
             _PM.set<int>("unbounded", MAX_BOUNDED_SEQ_SIZE);
         }
```

Comment:
We were limiting the use of unbounded to "large data", this is wrong, we should be able to test across all sizes with the -unbounded option.